### PR TITLE
Boiler house integration

### PR DESCRIPTION
**Pull request by LBHCallumM**
[Ticket](https://hackney.atlassian.net/browse/MR-716?atlOrigin=eyJpIjoiYzI1M2RjZTJjNTlhNGY4ZWEyZTU2NTFjNTY2OTU4MmYiLCJwIjoiaiJ9)

## Summary of Changes

- Added boiler house details to the sidebar on the property page. 
- Added form to add a boilerHouse to a property (via housingSearch API)
- Added a confirmation modal to boilerHouseDetails component when removing a boilerHouse from a property 
- Updated `webpack-cli` to 4.10.0 due to failing builds

I have used a lot of custom hooks, as this helps simplify the logic into smaller, more specific, uses. If you find this approach harder to read, I will consider any refactoring suggestions to improve readability for all developers. 

### Boiler house in the property sidebar
image

### No boiler house in the property sidebar

image

### Confirmation message when removing boiler house from property

image

### Add boiler house form

image

image

image

image

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -60,7 +60,7 @@
     "ts-config-single-spa": "3.0.0",
     "typescript": "^4.3.4",
     "webpack": "5.53.0",
-    "webpack-cli": "4.9.0",
+    "webpack-cli": "4.10.0",
```
Comment (ange47rm) [inline package.json:63]: Why was this updated?

File: src/views/add-boiler-house-view/add-boiler-house-view.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+import { useAsset } from "@mtfh/common/lib/api/asset/v1";
+import { Center, Spinner } from "@mtfh/common/lib/components";
+
+export const AddBoilerHouseView = (): JSX.Element => {
+  const { assetId } = useParams<{ assetId: string }>();
+
+  const { data: asset } = useAsset(assetId);
+
+  if (asset === undefined) {
```
Comment (ange47rm) [inline src/views/add-boiler-house-view/add-boiler-house-view.tsx:14]: What if the asset is not retrieved at all? Unlikely but possible

File: src/components/asset-sidebar/index.tsx [hunk 3]
```diff
@@ -19,6 +20,17 @@ interface Props extends Partial<SideBarProps> {
   showCautionaryAlerts: boolean;
 }
 
+const boilerHouseAssetTypes = new Set<string>([
```
Comment (ange47rm) [inline src/components/asset-sidebar/index.tsx:23]: Nice

File: src/components/add-boiler-house-form/utils.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { axiosInstance } from "@mtfh/common";
+import { Asset } from "@mtfh/common/lib/api/asset/v1";
+import { config } from "@mtfh/common/lib/config";
+
+export interface PatchAssetRequest {
+  boilerHouseId: string;
+}
+
+export const patchAsset = async (
```
Comment (ange47rm) [inline src/components/add-boiler-house-form/utils.ts:9]: Isn't this call set up in Common MFE?

File: src/components/add-boiler-house-form/add-boiler-house-form.tsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        </div>
+      </form>
+
+      {loading ? (
+        <Center>
+          <Spinner />
+        </Center>
+      ) : (
+        <>
+          {touched && (
```
Comment (ange47rm) [inline src/components/add-boiler-house-form/add-boiler-house-form.tsx:81]: We have a component called InlineSearchForm which is also used in New Asset form and I believe it has a similar function. Can that be used here maybe?

**Comment by ange47rm** [comment]
Nice work. Abstraction is good but I think I wouldn't have created the `useConfirmRemoveModal` hook at all. Also all 4 hooks are very specific, meaning these hooks are unlikely going to be re-used elsewhere. I prefer the utils/helper file for abstraction in these cases, however the code is readable, so while I don't fully agree on the use of hooks I'm happy to accept the PR once all other comments have been replied to 👍

**Comment by LBHCallumM** [comment]
> Nice work. Abstraction is good but I think I wouldn't have created the `useConfirmRemoveModal` hook at all. Also all 4 hooks are very specific, meaning these hooks are unlikely going to be re-used elsewhere. I prefer the utils/helper file for abstraction in these cases, however, the code is readable, so while I don't fully agree on the use of hooks I'm happy to accept the PR once all other comments have been replied to 👍

Why do we still think a hook needs to be reusable? You wouldn't make the same argument if I created a new class in an .NET API. A hook is an alternative to a component when there is no UI.

I'm not opposed to putting any logic into a utils file. However, that doesn't allow us to manage state, which would defeat the purpose.

**Comment by ange47rm** [comment]
> > Nice work. Abstraction is good but I think I wouldn't have created the `useConfirmRemoveModal` hook at all. Also all 4 hooks are very specific, meaning these hooks are unlikely going to be re-used elsewhere. I prefer the utils/helper file for abstraction in these cases, however, the code is readable, so while I don't fully agree on the use of hooks I'm happy to accept the PR once all other comments have been replied to 👍
> 
> Why do we still think a hook needs to be reusable? You wouldn't make the same argument if I created a new class in an .NET API. A hook is an alternative to a component when there is no UI.
> 
> I'm not opposed to putting any logic into a utils file. However, that doesn't allow us to manage state, which would defeat the purpose.

We had this debate before 😄 We just have a different view on it and like I said the code is readable so I'm happy to approve the PR. If you could reply to my other comments first that would be great

File: src/views/add-boiler-house-view/add-boiler-house-view.tsx (hunk 2 above)
Comment (LBHCallumM) [reply src/views/add-boiler-house-view/add-boiler-house-view.tsx:14]: Youre right, I will add better error handling

File: src/components/add-boiler-house-form/utils.ts (hunk 4 above)
Comment (LBHCallumM) [reply src/components/add-boiler-house-form/utils.ts:9]: Yeah, I must have forgotten to change the import

File: src/components/add-boiler-house-form/add-boiler-house-form.tsx (hunk 5 above)
Comment (LBHCallumM) [reply src/components/add-boiler-house-form/add-boiler-house-form.tsx:81]: I forgot about that component. Im not sure if its worth changing now

**Review by ange47rm — APPROVED** [review]

**Pull request merged**